src/jit/lclvars.cpp

```diff
@@ -108,12 +108,19 @@ void Compiler::lvaInitTypeRef()
     /* Set compArgsCount and compLocalsCount */
 
     info.compArgsCount = info.compMethodInfo->args.numArgs;
+    unsigned   argRegCount      = 0;
+
+    auto incrementArgCount = [this, &argRegCount]()
+    {
+        ++info.compArgsCount;
+        ++argRegCount;
+    };
 
```

Comment:
Isn't this going to double-count the "specials"? Don't they have varDsc that is going to be counted by your new loop below the call to lvaInitArgs()?